File: frontend/src/types/notification.ts
export type NotificationType =
  | "match_found"
  | "match_confirmed"
  | "item_claimed"
  | "item_closed"
  | "system";

export interface AppNotification {
  id: string;
  type: NotificationType;
  title: string;
  body: string;
  is_read: boolean;
  item_id: string | null;
  match_id: string | null;
  created_at: string;
}

export interface NotificationQuery {
  unread_only?: boolean;
  page?: number;
  page_size?: number;
}

File: backend/app/schemas/notification.py
"""Notification schemas (DTOs)."""

from __future__ import annotations

import datetime as dt
import uuid

from pydantic import BaseModel, ConfigDict

from app.models.notification import NotificationType


class NotificationRead(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: uuid.UUID
    type: NotificationType
    title: str
    body: str
    is_read: bool
    item_id: uuid.UUID | None
    match_id: uuid.UUID | None
    created_at: dt.datetime


class NotificationListResponse(BaseModel):
    items: list[NotificationRead]
    total: int
    page: int
    page_size: int
    total_pages: int


class UnreadCountResponse(BaseModel):
    count: int
